File: level 14/homework/homework.py

# ბულეანი (Boolean) არის მონაცემთა ტიპი, რომელიც მხოლოდ ორი მნიშვნელობით იწერება: ჭეშმარიტი (True) და ცრუ (False). ის ხშირად გამოიყენება ლოგიკურ გამოთვლებში და პროგრამირებაში, რათა განსაზღვროს, კონკრეტული პირობა შესრულებულია თუ არა.

# ბინარული კოდი (Binary Code) არის ინფორმაცია, რომელიც იწერება მხოლოდ ორ სიმბოლოთი: 0 და 1. ეს სიმბოლოები წარმოადგენს ბინარული ციფრების (ბიტების) ორ სხვადასხვა მდგომარეობას. ბინარული კოდი გამოიყენება კომპიუტერში მონაცემთა შესანახად და გადაცემისას, რადგან კომპიუტერები მუშაობენ ბინარული ლოგიკის საფუძველზე.


5 > 3
10 > 7
20 > 15
100 > 50
3.5 > 2.5
-1 > -5
8 > 2
25 > 20
500 > 300
7 > 4


3 < 5
7 < 10
15 < 20
50 < 100
2.5 < 3.5
-5 < -1
2 < 8
20 < 25
300 < 500
4 < 7


5 == 5
10 == 10
20 == 20
100 == 100
3.5 == 3.5
-1 == -1
8 == 8
25 == 25
500 == 500
7 == 7


5 != 3
10 != 7
20 != 15
100 != 50
3.5 != 2.5
-1 != -5
8 != 2
25 != 20
500 != 300
7 != 4


# უფრო მეტი (>): ეს ოპერატორი შეადარებს ორ მნიშვნელობას და დააბრუნებს ჭეშმარიტ (True) მნიშვნელობას, თუ მარცხენა მნიშვნელობა უფრო დიდია მარჯვენაზე. მაგალითად, 5 > 3 დააბრუნებს ჭეშმარიტს (True), რადგან 5 უფრო დიდია 3-ზე.

# ნაკლები (<): ეს ოპერატორი შეადარებს ორ მნიშვნელობას და დააბრუნებს ჭეშმარიტს (True) მნიშვნელობას, თუ მარცხენა მნიშვნელობა ნაკლებია მარჯვენაზე. მაგალითად, 3 < 5 დააბრუნებს ჭეშმარიტს (True), რადგან 3 ნაკლებია 5-ზე.

# თანასწორი (==): ეს ოპერატორი შეადარებს ორ მნიშვნელობას და დააბრუნებს ჭეშმარიტს (True) მნიშვნელობას, თუ ორივე მნიშვნელობა ერთმანეთთან ტოლია. მაგალითად, 5 == 5 დააბრუნებს ჭეშმარიტს (True), რადგან ორივე მნიშვნელობა ტოლია.

# არათანასწორი (!=): ეს ოპერატორი შეადარებს ორ მნიშვნელობას და დააბრუნებს ჭეშმარიტს (True) მნიშვნელობას, თუ ორივე მნიშვნელობა ერთმანეთთან არათოლია. მაგალითად, 5 != 3 დააბრუნებს ჭეშმარიტს (True), რადგან 5 და 3 არატოლია.

num = int(input("Enter a number: "))
num2 = int(input("Enter a number: "))

print(num > num2)
print(num < num2)
print(num == num2)
print(num != num2)

File: level 79/homework/homework.js
// 1. Copy Object Keys to Array
const obj = { name: 'John', age: 30 };
const keys = [];
for (let key in obj) {
  keys.push(key);
}
console.log('1. Object Keys:', keys);

// 2. Sum Only Numeric Properties
const data = { a: 10, b: 'hi', c: 20 };
let sum = 0;
for (let key in data) {
  if (typeof data[key] === 'number') {
    sum += data[key];
  }
}
console.log('2. Total Sum:', sum);

// 3. Convert Object to Key=Value String
const person = { name: 'John', age: 30, city: 'Paris' };
let str = '';
for (let key in person) {
  str += `${key}=${person[key]}, `;
}
str = str.slice(0, -2);
console.log('3. Key=Value String:', str);

// 4. List All Keys with Their Types
const user = { name: 'Alice', age: 25, active: true };
console.log('4. Keys and Types:');
for (let key in user) {
  console.log(`${key}: ${typeof user[key]}`);
}

// 5. Detect Nested Object
const complex = { name: 'Bob', address: { city: 'Rome' }, age: 40 };
for (let key in complex) {
  if (typeof complex[key] === 'object' && complex[key] !== null) {
    console.log(`5. Nested object found at key: ${key}`);
  }
}

// 6. Filter Long Words
const words = ['apple', 'elephant', 'cat', 'strawberry'];
console.log('6. Words longer than 5 characters:');
for (let word of words) {
  if (word.length > 5) {
    console.log(word);
  }
}

// 7. Create a String from Characters
const chars = ['H', 'e', 'l', 'l', 'o'];
let word = '';
for (let char of chars) {
  word += char;
}
console.log('7. Joined Word:', word);

// 8. Multiply Numbers Until Over 1000
let num = 1;
console.log('8. Multiplying numbers:');
do {
  console.log(num);
  num *= 2;
} while (num <= 1000);

// 9. Collect User Inputs (for browser use)
const numbers = [];
let input;
do {
  input = prompt("Enter a number (type 'stop' to finish):");
  if (input !== 'stop') {
    numbers.push(Number(input));
  }
} while (input !== 'stop');
console.log('9. Collected Inputs:', numbers);

// 10. Countdown Timer with Step
let count = 30;
console.log('10. Countdown:');
do {
  console.log(count);
  count -= 3;
} while (count >= 0);